Added explanation for chunk distribution (#14)
mqtt-varroa/varroa-documentation: LaTeX/Parts/Chapter5.tex

Every Agent holds a number of MQTT Agents and each MQTT Agent manages one MQTT Client.
\newpage

-\section{Chunk Concept}
A chunk is a work package of the scenario execution designed to be distributed from the Commander to its Agents.
It is then handled by the Agent it is distributed to.

> The Agent now creates or uses existing MQTT Agents to execute the Chunk.

Reviewer: ```suggestion
The Agent then creates or uses existing MQTT Agents to execute the Chunk.
```

It contains:
\begin{itemize}

> 	\item \textbf{clientCountInChunk}: the maximum amount of clients a chunk can represent.

Reviewer: maximum is wrong. It IS the amount of clients
```suggestion
	\item \textbf{clientCountInChunk}: the amount of clients the chunk represents.
```

	\item \textbf{clientOffset}: starting index of the clients contained in the chunk.

> 	\item \textbf{clientGroupId}: identfier for the Client Group contain in the chunk.

Reviewer: ```suggestion
	\item \textbf{clientGroupId}: identfier for the Client Group contained in the chunk.
```

> 	\item \textbf{clientGroupInformation}: contains information concerning the Client Group and MQTT-specific properties.

Reviewer: I would remove "and MQTT-specific properties"
```suggestion
	\item \textbf{clientGroupInformation}: contains information concerning the Client Group.
```

	\item \textbf{stageDuration}: the duration of the stage this chunk belongs to.

> 	\item \textbf{lifecycleRate}: the rate at which the contained commands are executed.

Reviewer: stageDuration and lifecycleRate were removed

	\item \textbf{commands}: a list of commands that are executed by the clients in the chunk.
\end{itemize}
%Due to the distributed nature of Varroa it was necessary to be able to spread the information contained in the scenario.
%To secure a deterministic distribution of a scenario's data the chunks had to be evenly spread across the initialized Agents.

\subsection{Chunk Distribution}
\begin{figure}[h]
	\begin{center}
	\includegraphics[scale=1.2]{Resources/PDF/ChunkDistribution}
	\caption{Distribution of a Chunk}
	\label{pic:ChunkDistribution}
	\end{center}
\end{figure}

> To secure a deterministic execution of a scenario the every chunk needs to be distributed to the same Agent in every execution of the scenario.

Reviewer: "every chunk needs to be distributed to the same Agent" is wrong.
```suggestion
A Chunk is deterministically assigned to an Agent to ensure that it gets executed by the same Agent even on multiple runs of the scenario. This determinism is needed to avoid fluctuation of reporting results that would be caused by Varroa and not the MQTT system under test.
```

> To ensure this determinism a round-robin distribution of the scenario's chunks combined with a handshake mechanism is used:

Reviewer: ```suggestion
To ensure this determinism a round-robin distribution of the scenario's Chunks combined with a handshake mechanism is used:
```

In accordance to figure \ref{pic:ChunkDistribution} the Commander tries to distribute a Chunk to an Agent who then answers with an accept or reject.
If the Agent rejects the Chunk (see figure \ref{pic:RejectHandshake}) the Commander then tries to distribute the Chunk to the next Agent in the same manner until one Agent accepts.

> When accepting the Chunk the Agents waits for a Start Message from the Commander and upon receiving it the Agent starts executing the Chunk.

Reviewer: ```suggestion
When accepting the Chunk the Agent waits for a Start Message from the Commander and upon receiving it starts executing the Chunk.
```

> After a successful execution of the Chunk the Agents sends a Finished Message to the Commander (see figure \ref{pic:AcceptHandshake}).

Reviewer: ```suggestion
After a successful execution of the Chunk the Agent sends a Finished Message to the Commander (see figure \ref{pic:AcceptHandshake}).
```
